setup.sh

```diff
@@ -84,7 +84,41 @@ project_id = "${OPS_PROJECT}"
 EOF
     terraform -chdir=${OPS_ENVIRONMENT_DIR} init -backend-config="bucket=${STATE_GCS_BUCKET_NAME}" -backend-config="prefix=ops"
     terraform -chdir=${OPS_ENVIRONMENT_DIR} apply --auto-approve
+fi # $SKIP_TERRAFORM
 
+####################
+# Build Containers #
+####################
+
+SETUP_IMAGE_TAG="setup"
+E2E_RUNNER_TAG="latest"
+
+if [[ -z "$SKIP_BUILD" ]]; then
+
+echo
+echo "$(tput bold)Building container images for testing and application hosting...$(tput sgr0)"
+echo
+
+gcloud builds submit "content-api" --async \
```

Comment:
Blocking Question: If the build fails, what does the user do? Presumably this script will not fail.

Reply:
Good question. If the build fails, the script actually will fail later on when it tries to deploy a Cloud Run service and it can't find the image.  So the user will still face an error from `setup.sh`, just a different kind of error. To know if the build failed, they would need to check the build log. 

We could make this more discoverable by returning the URL to the build job after it's started but I think that could be tracked in a separate feature request.

Reply:
Yes, I definitely think we should link to the build url.

Let me expand a bit on my concerns around user impact:

1. We're racing the build against the script between here and when gcloud run deploy goes to pull the image. Off the top of my head, image build & push can take 20-90s. First Cloud Run deploy can take 30-90s. Maybe my numbers are off: have you done a few test throughs to see if this script consistently works as expected?

2. We don't fail fast, we fail late: a Cloud Run deploy failure should route people back to re-running setup.sh. If someone needs to troubleshoot, the error won't be targeted to the right place in setup.sh, so they'll need to infer from the Cloud Run error what to do.

Idea for (1):  If the numbers are too tight, my thought is maybe the e2e runner and api should be built async first, and the website kept as "sync" with quiet output. The website build time can then be a rough marker on both images being ready, for a more reliable race. Backup idea: we add a poll on Cloud Build and block deploy until the image is ready.

Reply:
The build commands are moved further up in the setup script and run while the application terraform runs.  In my tests, this gives the build jobs ample time to complete.  

I didn't plan for any build job failures or delays.  If anything, I think adding a check for the build before each corresponding run command with a couple of retries could cover these concerns. However I think it should be part of a separate PR, as well as other opportunities to improve error handling of the set up script. 

I think that if we can feel confident that this setup script completes under normal circumstances, we can merge this before the error handling.

Reply:
Rather than implementing retries, I added a function that will simply wait for the build status to move past 'WORKING' before deploying the Run service.  

1) We won't have to race the build against the script.
2) If someone encounters an error due to build, they will be presented with a link to the logs, as well as advice to re-run setup.  The error will be targeted at a more accurate place in setup.sh